static/js/services/enketo.js

```diff
@@ -432,14 +432,21 @@ angular.module('inboxServices').service('Enketo',
     };
 
     var saveDocs = function(docs) {
-      return $q.all(docs.map(function(doc) {
-        return DB()
-          .put(doc)
-          .then(function(res) {
-            doc._rev = res.rev;
-            return doc;
+      return DB().bulkDocs(docs)
```

Comment:
So to be clear, this *isn't* a save where order is necessary?

Reply:
The order is already correct. The main doc is first in the array, and all the other docs are later.